File: MiniDOMOffice/Office/GDE/Templates/Items/TemplateItemsCollection.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class Office
    {

        /// <summary>
        /// Collezione di elementi generati da un template
        /// </summary>
        [Serializable]
        public class TemplateItemsCollection
            : CCollection<TemplateItem>
        {
           
            [NonSerialized] private DocumentTemplate m_Document;

            /// <summary>
            /// Costruttore
            /// </summary>
            public TemplateItemsCollection()
            {
                m_Document = null;
            }

            /// <summary>
            /// Costruttore
            /// </summary>
            /// <param name="doc"></param>
            public TemplateItemsCollection(DocumentTemplate doc)
                : this()
            {
                if (doc is null)
                    throw new ArgumentNullException("document");
                m_Document = doc;
            }

            /// <summary>
            /// Imposta il template
            /// </summary>
            /// <param name="doc"></param>
            protected internal virtual void SetDocument(DocumentTemplate doc)
            {
                m_Document = doc;
            }

            /// <summary>
            /// Aggiunge un elemento nel template
            /// </summary>
            /// <param name="type"></param>
            /// <param name="text"></param>
            /// <param name="bounds"></param>
            /// <returns></returns>
            public TemplateItem Add(TemplateItemTypes type, string text, CRectangle bounds)
            {
                var ret = new TemplateItem(type, text, bounds);
                Add(ret);
                return ret;
            }

            /// <summary>
            /// Aggiunge un nuovo elemento al template
            /// </summary>
            /// <param name="type"></param>
            /// <param name="text"></param>
            /// <param name="bounds"></param>
            /// <param name="color"></param>
            /// <returns></returns>
            public TemplateItem Add(TemplateItemTypes type, string text, CRectangle bounds, string color)
            {
                var ret = new TemplateItem(type, text, bounds, color);
                Add(ret);
                return ret;
            }
        }
    }
}

File: MiniDOMOffice/Office/GDE/Templates/TemplateItemsCollection.vb
Partial Public Class Office

    <Serializable>
    Public Class TemplateItemsCollection
        Inherits CCollection(Of TemplateItem)

        Private m_Document As DocumentTemplate

        Public Sub New()
            Me.m_Document = Nothing
        End Sub

        Public Sub New(ByVal doc As DocumentTemplate)
            Me.New
            If (doc Is Nothing) Then Throw New ArgumentNullException("document")
            Me.m_Document = doc
        End Sub

        Protected Friend Overridable Sub SetDocument(ByVal doc As DocumentTemplate)
            Me.m_Document = doc
        End Sub

        Public Overloads Function Add(ByVal type As TemplateItemTypes, ByVal text As String, ByVal bounds As CRectangle) As TemplateItem
            Dim ret As New TemplateItem(type, text, bounds)
            Me.Add(ret)
            Return ret
        End Function

        Public Overloads Function Add(ByVal type As TemplateItemTypes, ByVal text As String, ByVal bounds As CRectangle, ByVal color As String) As TemplateItem
            Dim ret As New TemplateItem(type, text, bounds, color)
            Me.Add(ret)
            Return ret
        End Function

    End Class

End Class